cmd/preflight/cmd/check_container.go

```diff
@@ -103,85 +107,131 @@ func checkContainerRunE(cmd *cobra.Command, args []string, runpreflight runPrefl
 		return fmt.Errorf("invalid configuration: %w", err)
 	}
 
-	artifactsWriter, err := artifacts.NewFilesystemWriter(artifacts.WithDirectory(cfg.Artifacts))
+	manifestBytes, err := crane.Manifest(containerImage)
 	if err != nil {
-		return err
+		return fmt.Errorf("could not retrieve manifest: %w", err)
 	}
-
-	// Add the artifact writer to the context for use by checks.
-	ctx = artifacts.ContextWithWriter(ctx, artifactsWriter)
-
-	formatter, err := formatters.NewByName(formatters.DefaultFormat)
+	manifest, err := cranev1.ParseManifest(bytes.NewBuffer(manifestBytes))
 	if err != nil {
-		return err
+		return fmt.Errorf("could not parse manifest: %w", err)
 	}
 
-	opts := generateContainerCheckOptions(cfg)
-
-	checkcontainer := container.NewCheck(
-		containerImage,
-		opts...,
-	)
+	platformChanged := cmd.Flags().Lookup("platform").Changed
 
-	pc := lib.NewPyxisClient(ctx, cfg.CertificationProjectID, cfg.PyxisAPIToken, cfg.PyxisHost)
-	resultSubmitter := lib.ResolveSubmitter(pc, cfg.CertificationProjectID, cfg.DockerConfig, cfg.LogFile)
+	containerImagePlatforms := []string{cfg.Platform}
 
-	// Run the  container check.
-	cmd.SilenceUsage = true
-
-	err = runpreflight(
-		ctx,
-		checkcontainer.Run,
-		cli.CheckConfig{
-			IncludeJUnitResults: cfg.WriteJUnit,
-			SubmitResults:       cfg.Submit,
-		},
-		formatter,
-		&runtime.ResultWriterFile{},
-		resultSubmitter,
-	)
+	// If platform param is not changed, it means that a platform was not specified on the
+	// command line. Therefore, we should process all platforms in the manifest list.
+	// As long as what is poinged to is a manifest list. Otherwise, it will just be the
+	// currnt runtime platform.
+	if isImageList(manifest.MediaType) && !platformChanged {
+		logger.V(log.DBG).Info("manifest list detected, checking all platforms in manifest")
+		manifest, err := cranev1.ParseIndexManifest(bytes.NewBuffer(manifestBytes))
+		if err != nil {
+			return fmt.Errorf("could not parse index manifest: %w", err)
+		}
 
-	if err != nil {
-		return err
+		// Look up the manifestListDigest and set it in the config
+		manifestListDigest, err := crane.Digest(containerImage)
+		if err != nil {
+			return fmt.Errorf("could not get manifest list digest")
```

Comment:
Do we want to wrap the error here as well?

Reply:
Good catch. Yes.